libs/vkd3d/command.c

```diff
@@ -1341,7 +1341,11 @@ static VkDescriptorPool d3d12_command_allocator_allocate_descriptor_pool(
     {
         pool_desc.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO;
         pool_desc.pNext = NULL;
-        pool_desc.flags = 0;
+        /* For a correct implementation of RS 1.0 we need to update packed descriptor sets late rather than on draw.
+         * If device does not support descriptor indexing, we must update on draw and pray applications don't rely on RS 1.0
+         * guarantees. */
+        pool_desc.flags = allocator->device->vk_info.EXT_descriptor_indexing ?
+                          VK_DESCRIPTOR_POOL_CREATE_UPDATE_AFTER_BIND_BIT_EXT : 0;
```

Comment:
We should probably check for *all* the `UpdateAfterBind` features separately, and store whether we support volatile descriptors somewhere in the `d3d12_device`. Older Nvidia generations (up to Pascal) do not support `descriptorBindingUniformBufferUpdateAfterBind`.